components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -265,6 +267,291 @@
+        } else {
+            for (ExpressionNode expressionNode : expressionNodes) {
+                String operation = expressionNode.getOperation();
+                String value = expressionNode.getValue();
+                String attributeName = expressionNode.getAttributeValue();
+                if (StringUtils.isNotBlank(attributeName) || StringUtils.isNotBlank(value) || StringUtils
+                        .isNotBlank(operation)) {
+                    switch (attributeName) {
+                        case IdPManagementConstants.IDP_NAME:
+                            attributeName = IdPManagementConstants.NAME;
+                            break;
+                        case IdPManagementConstants.IDP_DESCRIPTION:
+                            attributeName = IdPManagementConstants.DESCRIPTION;
+                            break;
+                        case IdPManagementConstants.IDP_HOME_REALM_ID:
+                            attributeName = IdPManagementConstants.HOME_REALM_ID;
+                            break;
+                        case IdPManagementConstants.IDP_IS_ENABLED:
+                            attributeName = IdPManagementConstants.IS_ENABLED;
+                            break;
+                        case IdPManagementConstants.IDP_UUID:
+                            attributeName = IdPManagementConstants.UUID;
+                            break;
+                        default:
+                            if (log.isDebugEnabled()) {
+                                log.debug("Invalid filter attribute name. Filter attribute : " + attributeName);
+                            }
+                            filterQueryBuilder.setFilterQuery(IdPManagementConstants.EMPTY_STRING);
+                            filterQueryBuilder.setEmptyFilterAttributeValue();
+                            return;
+                    }
+                    if (IdPManagementConstants.EQ.equals(operation)) {
+                        filter.append(attributeName).append(" = ? AND ");
+                        filterQueryBuilder.setFilterAttributeValue(value);
+                    } else if (IdPManagementConstants.SW.equals(operation)) {
+                        filter.append(attributeName).append(" like ? AND ");
+                        filterQueryBuilder.setFilterAttributeValue(value + "%");
+                    } else if (IdPManagementConstants.EW.equals(operation)) {
+                        filter.append(attributeName).append(" like ? AND ");
+                        filterQueryBuilder.setFilterAttributeValue("%" + value);
+                    } else if (IdPManagementConstants.CO.equals(operation)) {
+                        filter.append(attributeName).append(" like ? AND ");
+                        filterQueryBuilder.setFilterAttributeValue("%" + value + "%");
+                    } else {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Invalid filter value. " + operation);
```

Comment:
let's throw a IdentityProviderManagementServerException if filter value is invalid